src/github.com/stellar/federation/request_handler.go

```diff
@@ -20,40 +19,26 @@ func (rh *RequestHandler) Main(w http.ResponseWriter, r *http.Request) {
 	q := r.URL.Query().Get("q")
 	switch {
 	case requestType == "name" && q != "":
-		rh.MakeDatabaseRequest(rh.config.FederationQuery, w, r)
+		rh.FedDBRequest(q, w)
 	case requestType == "id" && q != "":
-		rh.MakeDatabaseRequest(rh.config.ReverseFederationQuery, w, r)
+		rh.RevFedDBRequest(q, w)
 	default:
 		http.Error(w, ErrorResponseString("invalid_request", "Invalid request"), http.StatusBadRequest)
 	}
 }
 
-func (rh *RequestHandler) MakeDatabaseRequest(query string, w http.ResponseWriter, r *http.Request) {
-	record := Record{}
+func (rh *RequestHandler) RevFedDBRequest(accountID string, w http.ResponseWriter) {
```

Comment:
I understand that separating this method into federation and reverse federation may look nicer but now 75% of the code of these two methods is the same (code duplication).

What do you think about changing this file to something like this (live coded on Github):
```go
func (rh *RequestHandler) Main(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json")

	requestType := r.URL.Query().Get("type")
	q := r.URL.Query().Get("q")
	switch {
	case requestType == "name" && q != "":
		rh.FederationRequest(w, r)
	case requestType == "id" && q != "":
		rh.ReverseFederationRequest(w, r)
	default:
		http.Error(w, ErrorResponseString("invalid_request", "Invalid request"), http.StatusBadRequest)
	}
}

func (rh *RequestHandler) FederationRequest(w http.ResponseWriter, r *http.Request) {
	q := r.URL.Query().Get("q")
	name := ""
	domain := ""

	if i := strings.Index(q, "*"); i >= 0 {
		name = q[:i]
		domain = q[i+1:]
	}

	if name == "" || domain != rh.config.Domain {
		http.Error(w, ErrorResponseString("not_found", "Incorrect domain"), http.StatusNotFound)
		return
	}

	record := MakeDatabaseRequest(rh.config.FederationQuery, name)
	rh.SendResponse(record)
}

func (rh *RequestHandler) ReverseFederationRequest(w http.ResponseWriter, r *http.Request) {
	id := r.URL.Query().Get("q")
	record := MakeDatabaseRequest(rh.config.ReverseFederationQuery, id)
	rh.SendResponse(record)
}

func (rh *RequestHandler) SendResponse(w http.ResponseWriter, record Record) {
	json, err := json.Marshal(record)

	if err != nil {
		http.Error(w, ErrorResponseString("server_error", "Server error"), http.StatusInternalServerError)
		return
	}

	w.Write(json)
}

func (rh *RequestHandler) MakeDatabaseRequest(query string, value string) record Record {
	err := rh.database.Get(&record, query, value)

	if err != nil {
		if err.Error() == "sql: no rows in result set" {
			log.Print("Federation record NOT found")
			http.Error(w, ErrorResponseString("not_found", "Account not found"), http.StatusNotFound)
		} else {
			log.Print("Server error: ", err)
			http.Error(w, ErrorResponseString("server_error", "Server error"), http.StatusInternalServerError)
		}
		return
	}

	log.Print("Federation record found")
	

	var usernameBuffer bytes.Buffer
	usernameBuffer.WriteString(record.Username)
	usernameBuffer.WriteString("*")
	usernameBuffer.WriteString(rh.config.Domain)

	record.Username = usernameBuffer.String()
	return
}

func ErrorResponseString(code string, message string) string {
	error := Error{Code: code, Message: message}
	json, _ := json.Marshal(error)
	return string(json)
}
```
Now, we have separate methods for federation and reverse federation and no code duplication.